src/shared/codelens/typescriptCodeLensProvider.ts

```diff
@@ -5,154 +5,115 @@
+            const debugConfig: NodeDebugConfiguration = {
+                type: 'node',
+                request: 'attach',
+                name: 'SamLocalDebug',
+                preLaunchTask: undefined,
+                address: 'localhost',
+                port: debugPort!,
+                localRoot: samProjectCodeRoot,
+                remoteRoot: '/var/task',
+                protocol: 'inspector',
+                skipFiles: [
+                    '/var/runtime/node_modules/**/*.js',
+                    '<node_internals>/**/*.js'
+                ]
+            }
 
-    public resolveCodeLens(
-        codeLens: vscode.CodeLens,
-        token: vscode.CancellationToken
-    ): vscode.ProviderResult<vscode.CodeLens> {
-        throw new Error('not implemented')
-    }
+            const localLambdaRunner: LocalLambdaRunner = new LocalLambdaRunner(
+                configuration,
+                args,
+                debugPort,
+                'nodejs8.10', // TODO: Remove hard coded value
+                toolkitOutputChannel,
+                processInvoker,
+                taskInvoker,
+                debugConfig,
+                samProjectCodeRoot
 
-    private generateConfigureCodeLens(
-        document: vscode.TextDocument,
-        range: vscode.Range,
-        handlerName: string,
-        workspaceFolder?: vscode.WorkspaceFolder
-    ) {
-        // Handler will be the fully-qualified name, so we also allow '.' despite it being forbidden in handler names.
-        if (/[^\w\-\.]/.test(handlerName)) {
-            throw new Error(
-                `Invalid handler name: '${handlerName}'. ` +
-                'Handler names can contain only letters, numbers, hyphens, and underscores.'
             )
-        }
-
-        const command = {
-            arguments: [workspaceFolder, handlerName],
-            command: 'aws.configureLambda',
-            title: localize('AWS.command.configureLambda', 'Configure')
-        }
-
-        return new vscode.CodeLens(range, command)
-    }
-
-    private generateLocalInvokeCodeLens(
-        document: vscode.TextDocument,
-        range: vscode.Range,
-        handlerName: string,
-        debug: boolean,
-        workspaceFolder: vscode.WorkspaceFolder,
-    ): vscode.CodeLens {
-        const title: string = debug ?
-            localize('AWS.codelens.lambda.invoke.debug', 'Debug Locally') :
-            localize('AWS.codelens.lambda.invoke', 'Run Locally')
 
-        const commandArgs: LambdaLocalInvokeArguments = {
-            document,
-            range,
-            handlerName,
-            debug,
-            workspaceFolder,
+            await localLambdaRunner.run()
         }
+    )
+}
 
-        const command: vscode.Command = {
-            arguments: [commandArgs],
-            command: 'aws.lambda.local.invoke',
-            title
+export const makeTypescriptCodeLensProvider =  (): vscode.CodeLensProvider => {
```

Comment:
We use this style:

```typescript
export function makeTypescriptCodeLensProvider(): vscode.CodeLensProvider {
    ...
}
```

Same comment throughout this changeset.

Reply:
✔️